### ISSUE TITLE
Unable to shuffle if the fields are not of same length

### ISSUE THREAD

**Issue by michael20at** [Unity-Technologies/ml-agents#1798]
Hm, tried ml-agents 0.7 with my new 2080ti (runs great on normal Tensorflow), 
updated Unity to the newest version 2018.3.7f1, training of 3D Balls example via cmd starts fine, but after a few episode (sometimes only a few, sometimes after 9) I get 

"Unable to shuffle if the fields are not of same length"!

What could be the problem?

**Comment by michael20at** [comment]
Hm, it kinda works if I increase the buffer size by ten in the config file, but I still don't know why?

**Comment by xiaomaogy** [comment]
This is kind of weird and unexpected, I don't see other people raising this issue, so I guess maybe delete and rerun and try to follow the basic guide step by step might help.

**Comment by Ina299** [comment]
I also experienced this issue.
I use ml-agents 0.6 on 1080ti.
`       
        batch_size:     1536
        beta:   0.006
        buffer_size:    15360
        epsilon:        0.17
        gamma:  0.995
        hidden_units:   512
        lambd:  0.9
        learning_rate:  0.00012
        max_steps:      5.0e15
        normalize:      True
        num_epoch:      4
        num_layers:     4
        time_horizon:   768
        summary_freq:   2000
        use_recurrent:  False
        use_curiosity:  False
`
I saw this phenomenon when I took agent' s life time too long.
So I guess agent should be Done() before buffer_size.

**Comment by ervteng** [comment]
Are you using a Numpy version greater than 1.14.1 by any chance? Newer versions of Numpy are known to have similar issues.

**Comment by rainysolar** [comment]
I have the same issue.
I use a 2070 card which only support CUDA 10.
CUDA 10 needs TensorFlow >=1.13.0.
And Numpy 1.14.1 is not support TensorFlow>=1.13.0

**Comment by yttiktak** [comment]
Same issue. Using Ubuntu Bionic, numpy v 1.16.2
Trying to track it down, seems a numpy array (advantages) changes shape from (N,) to (1,N) when the ppy trainer.py subtracts the mean from it, about line 325. 
 I hacked out a thing to bypass policy updates when that happens. Walker seems to be training now. My 5 yr old approves of the walker. It got up to a reward of 300 so far.

**Comment by JadenTravnik** [comment]
This doesnt seem to happen when using Numpy 1.14.5 and tensorflow 1.7 as mentioned in [setup.py](https://github.com/Unity-Technologies/ml-agents/blob/37d139af636e4a2351751fbf0f2fca5a9ed7457f/ml-agents/setup.py#L33).
I'm still not sure what the issue is but at least one doesn't get the bug.

**Comment by EfveZombie** [comment]
It should be a compatibility problem with higher version of Tensorflow or Numpy.

In my situation (py 3.7 && tf 1.13 && np 1.16), simply adding parameter `dtype = float` to all the calls of `numpy.array()` in [ml-agents/mlagents/trainers/buffer.py](https://github.com/Unity-Technologies/ml-agents/blob/master/ml-agents/mlagents/trainers/buffer.py) could fix this problem. 

But I dont know if that would cause any potential problems.

**Comment by ervteng** [comment]
The fields in the buffer should be all numerical, so I don't see an issue with adding the `dtype`. We'll definitely track this as we upgrade to newer versions of Numpy.

**Comment by ervteng** [comment]
Hi @EfveZombie, we found the root cause of this issue and fixed it on the latest `develop` branch. Let us know if it fixes your issue. Thanks!

**Comment by awjuliani** [comment]
Closing this issue as it was addressed in a recent release of ML-Agents.
